src/gui/qgstemporalcontrollerwidget.cpp

```diff
@@ -145,6 +167,38 @@ void QgsTemporalControllerWidget::keyPressEvent( QKeyEvent *e )
   QWidget::keyPressEvent( e );
 }
 
+void QgsTemporalControllerWidget::aboutToShowRangeMenu()
+{
+  QgsDateTimeRange projectRange;
+  if ( QgsProject::instance()->timeSettings() )
+    projectRange = QgsProject::instance()->timeSettings()->temporalRange();
+  mRangeSetToProjectAction->setEnabled( projectRange.begin().isValid() && projectRange.end().isValid() );
+
+  mRangeLayersSubMenu->clear();
+  for ( int i = 0; i < mMapLayerModel->rowCount(); ++i )
+  {
+    QModelIndex index = mMapLayerModel->index( i, 0 );
+    QgsMapLayer *currentLayer = mMapLayerModel->data( index, QgsMapLayerModel::LayerRole ).value<QgsMapLayer *>();
+    if ( !currentLayer->temporalProperties() || !currentLayer->temporalProperties()->isActive() )
```

Comment:
We should probably add a filter for temporal layers to the map layer proxy model at some stage